.github/workflows/build-lint-test.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint on Python 3.11.
+# This initiates on every push or pull_request to develop
+
+name: Build, Lint, and Test
+
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
```

Comment:
Happy to go with Flake8 for now - though I've been wondering about trying out Ruff.  Have you had any experience with it?

Reply:
I agree with the suggestion. Ruff seems like a good updated option, joined with its extremely fast run time (perks of anything written in Rust). I have updated the action to use Ruff. At the moment, it is using its default configuration and treating all errors as warnings, so that the action completes and simply displays them.

There are several improvements that can be implemented, however these are at personal discretion so I've left them here as options:

1. The specific flakes, errors and warnings being check for, as well as all other Ruff config can be manually set in pyproject.toml and it will pick them up automatically.
2. The action of ruff check can be split into two, where the first one selects only a few specific rules to be checked for and make the action error if these are found (for more severe errors such as syntax errors), and then having a second check for all other desired rules that only serves as warnings (as the current setup does).
3. A few quick lines of code can be added for Ruff to automatically fix the errors found (those it can fix automatically) and commit these either to the codebase or the pull request, depending on the situation.

Reply:
Thanks for the suggestions.  Maybe you could make an issue to capture this into the future?  Or I can copy your text (making it clear you are the author) into an issue if you prefer.  I'd like to live with Ruff for a bit in its most basic form, before trying anything more sophisticated with it.

Reply:
Of course, I'll capture this in a new issue.